### ISSUE TITLE
Repeat flags

### ISSUE THREAD

**Issue by deladriere** [ptwz/python_wizard#4]
analyzing sounds from some old TMS5220 from the Talkie library shows lots of repeats.
I like them because my instruments use them a looping point.

I noticed that the repeat flag is never generated when I process sound with BlueWizardor python_wizard 
Is it on purpose or it's just not yet implemented?

**Comment by ptwz** [comment]
Hmm, i'm not quite sure why they have been omitted. I went for a straight port, honestly mostly transcribing code without thinking too much about it.
Actually I have to read up on repeats and how they actually work, maybe you could give some insight. As far as I understood from the TMS emulator library, they just tell the chip to use basically the same excitation and filter parameters as in the last frame, right?

**Comment by deladriere** [comment]
Yes, that's all I know about the repeat flag: it was made to save some memory.
I also checked some LPC stream I made with [QBOXpro: ](http://www.polaxis.be/2015/12/lpc-encoding-for-the-arduinos-talkie-library/) and they also lack the repeat flag. 
Have also look [here:](http://www.technozone51.fr/BlocklyDuinotest/speech.php) more than 100.000 french words encoded with QBOXpro and they also lack the repeat flag.
I didn't look inside your code yet, but is there a place where it calculates the repeat flag?

**Comment by ptwz** [comment]
Well, currently there is neither repeat code in BW nor in PW. I guess I could implement it, if needed. Currently my day job takes quite some time, but I hope to get around to it..

**Comment by ptwz** [comment]
Tried something out in  branch repeat_estimator, not finished yet though.

**Comment by toneburst** [comment]
Funnily enough, I have a use case for LPC files without either Repeat or Silent flags, so the current behaviour is great for me (at least for Repeat frames).

Could you make sure the current no-repeat-flag setup remains as an option, if you do add Repeat frame support?

Could you confirm that the Silent flag is also not supported? If it is, could you possible add an option to add silent frames (Amp and all “K” values zeroed?), instead of adding Silent frame flags to the output datastream?

Thanks very much for your work on this project! I’m so glad to have found it!!

**Comment by toneburst** [comment]
Funnily enough, I have a use case for LPC files without either Repeat or Silent flags, so the current behaviour is great for me (at least for Repeat frames).

Could you make sure the current no-repeat-flag setup remains as an option, if you do add Repeat frame support?

Could you confirm that the Silent flag is also not supported? If it is, could you possible add an option to add silent frames (Amp and all “K” values zeroed?), instead of adding Silent frame flags to the output datastream?

Thanks very much for your work on this project! I’m so glad to have found it!!

**Comment by toneburst** [comment]
Sorry, I didn’t notice how old this thread is. You’ve probably moved on to other things now. Completely understand if this is the case!
